src/query/functions/src/scalars/boolean.rs

```diff
@@ -338,7 +341,109 @@ pub fn register(registry: &mut FunctionRegistry) {
                     );
             }
             _ => unreachable!(),
-        });
+        })
+    }
+
+    for src_type in ALL_FLOAT_TYPES {
+        with_float_mapped_type!(|NUM_TYPE| match src_type {
+            NumberDataType::NUM_TYPE => {
+                registry.register_1_arg::<NumberType<NUM_TYPE>, BooleanType, _, _>(
+                    "to_boolean",
+                    |_, domain| {
+                        FunctionDomain::Domain(BooleanDomain {
+                            has_false: domain.min <= OrderedFloat(0.0)
+                                && domain.max >= OrderedFloat(0.0),
+                            has_true: !(domain.min == OrderedFloat(0.0)
+                                && domain.max == OrderedFloat(0.0)),
+                        })
+                    },
+                    |val, _| val != OrderedFloat(0.0),
+                );
+
+                registry
+                    .register_combine_nullable_1_arg::<NumberType<NUM_TYPE>, BooleanType, _, _>(
+                        "try_to_boolean",
+                        |_, domain| {
+                            FunctionDomain::Domain(NullableDomain {
+                                has_null: false,
+                                value: Some(Box::new(BooleanDomain {
+                                    has_false: domain.min <= OrderedFloat(0.0)
+                                        && domain.max >= OrderedFloat(0.0),
+                                    has_true: !(domain.min == OrderedFloat(0.0)
+                                        && domain.max == OrderedFloat(0.0)),
+                                })),
+                            })
+                        },
+                        vectorize_with_builder_1_arg::<
+                            NumberType<NUM_TYPE>,
+                            NullableType<BooleanType>,
+                        >(|val, output, _| {
+                            output.builder.push(val != OrderedFloat(0.0));
+                            output.validity.push(true);
+                        }),
+                    );
+
+                let name = format!("to_{src_type}").to_lowercase();
```

Comment:
seems weird. you can rename `src_type` to `float_ty`

Reply:
we have float32 and float64, can't be sure which one is the source type.